src/rgw/rgw_quota.cc

```diff
@@ -232,9 +232,29 @@ class RGWQuotaStatsUpdate : public lru_map<T, RGWQuotaCacheStats>::UpdateContext
     const uint64_t rounded_added = rgw_rounded_objsize(added_bytes);
     const uint64_t rounded_removed = rgw_rounded_objsize(removed_bytes);
 
-    entry->stats.size += added_bytes - removed_bytes;
-    entry->stats.size_rounded += rounded_added - rounded_removed;
-    entry->stats.num_objects += objs_delta;
+    if ((added_bytes - removed_bytes) < 0) {
```

Comment:
yes, this check is also valid, would change it this way if this is more readable.